cmd/shell.go

```diff
@@ -188,19 +199,57 @@ func (as *AppShell) VersionString() string {
 func FailOnError(err error, msg string) {
 	if err != nil {
 		// AUDIT[ Error Conditions ] 9cc4d537-8534-4970-8665-4b382abe82f3
-		panic(fmt.Sprintf("%s: %s", msg, err))
+		fmt.Fprintf(os.Stderr, "%s: %s", msg, err)
+		os.Exit(1)
 	}
 }
 
 // AmqpChannel is the same as amqpConnect in boulder, but with even
 // more aggressive error dropping
-func AmqpChannel(url string) (ch *amqp.Channel) {
-	conn, err := amqp.Dial(url)
-	FailOnError(err, "Unable to connect to AMQP server")
+func AmqpChannel(conf Config) (*amqp.Channel, error) {
+	var conn *amqp.Connection
 
-	ch, err = conn.Channel()
-	FailOnError(err, "Unable to establish channel to AMQP server")
-	return
+	if conf.AMQP.SSL != nil {
+		if strings.HasPrefix(conf.AMQP.Server, "amqps") == false {
+			err := fmt.Errorf("SSL configuration provided, but not using an AMQPS URL")
+			return nil, err
+		}
+
+		cfg := new(tls.Config)
+
+		cert, err := tls.LoadX509KeyPair(conf.AMQP.SSL.CertFile, conf.AMQP.SSL.KeyFile)
```

Comment:
I was really hoping that there'd be semantics like `getString(key, default)` in the [most popular Golang](https://github.com/spf13/viper) configuration systems, but it looks like they're all about the nil-checks as well.

Opened issue #387.